rr fails to compile on RHEL7

When @wcohen tried to compile rr on RHEL7, the build failed because `memfd_create` is not available on RHEL7. The use of this function was introduced in commit a6fb7f1db1564e0d85f8c5da19bec8c178e56774.

```
make[2]: Entering directory `/home/wcohen/rpmbuild/BUILD/rr-7908feaee2ddedcee6db7619a221b24193fb5e99/obj'
[ 21%] Building CXX object CMakeFiles/rr.dir/src/EmuFs.cc.o
    88 |   ScopedFd fd = memfd_create(real_name.c_str(), 0);  
       |                 ^~~~~~~~~~~~  
       |                 timer_create  
make[2]: *** [CMakeFiles/rr.dir/src/EmuFs.cc.o] Error 1
make[2]: Leaving directory `/home/wcohen/rpmbuild/BUILD/rr-7908feaee2ddedcee6db7619a221b24193fb5e99/obj'
make[1]: *** [CMakeFiles/rr.dir/all] Error 2
make[1]: Leaving directory `/home/wcohen/rpmbuild/BUILD/rr-7908feaee2ddedcee6db7619a221b24193fb5e99/obj'
make: *** [all] Error 2
```
It was suggested that we make use of some `ifdefs` to adjust things if `memfd_create` is not available.

--- a/src/EmuFs.cc
+++ b/src/EmuFs.cc
@@ -85,7 +85,7 @@ static ScopedFd create_memfd_file(const string& orig_path, dev_t orig_device,
        << "-inode-" << orig_inode << "-" << orig_path;
   real_name = name.str().substr(0, 255);
 
-  ScopedFd fd = memfd_create(real_name.c_str(), 0);
+  ScopedFd fd = syscall(SYS_memfd_create, real_name.c_str(), 0);
   return fd;
 }
 

